status/k8s.go

```diff
@@ -477,12 +480,66 @@ func (k *K8sStatusCollector) status(ctx context.Context) *Status {
 		},
 	}
 
+	// for the sake of sanity, don't get pod logs more than once
+	gotPodLogs := false
 	err := k.podStatus(ctx, status, defaults.AgentDaemonSetName, "k8s-app=cilium", func(ctx context.Context, status *Status, name string, pod *corev1.Pod) {
 		if pod.Status.Phase == corev1.PodRunning {
+			// extract container status
+			var containerStatus *corev1.ContainerStatus
+			for i, cStatus := range pod.Status.ContainerStatuses {
+				if cStatus.Name == defaults.AgentContainerName {
+					containerStatus = &pod.Status.ContainerStatuses[i]
+					break
+				}
+			}
 			tasks = append(tasks, statusTask{
 				name: pod.Name,
-				task: func(_ context.Context) error {
-					s, err := k.client.CiliumStatus(ctx, k.params.Namespace, pod.Name)
+				task: func(ctx context.Context) error {
+					var s *models.StatusResponse
+					var err error
+
+					if containerStatus != nil && containerStatus.State.Running != nil {
+						// if container is running, execute "cilium status" in the container and parse the result
+
+						s, err = k.client.CiliumStatus(ctx, k.params.Namespace, pod.Name)
+
+					} else {
+						// otherwise, generate a useful status message
+						desc := "is not running"
+						lastLog := fmt.Sprintf("try 'kubectl -n %s logs -c %s %s'",
+							pod.Namespace, defaults.AgentContainerName, pod.Name)
+
+						// determine CrashLoopBackOff status. Yes, it is this ugly.
+						if containerStatus != nil {
+							if containerStatus.State.Waiting != nil && containerStatus.State.Waiting.Reason == "CrashLoopBackOff" {
+								desc = "is in CrashLoopBackOff"
+							}
+							if containerStatus.LastTerminationState.Terminated != nil {
+								terminated := containerStatus.LastTerminationState.Terminated
+								desc = fmt.Sprintf("%s, exited with code %d", desc, terminated.ExitCode)
+
+								// capture final log line, maybe it's useful
+								// either from container message or
+								dyingGasp := ""
+								if terminated.Message != "" {
+									dyingGasp = terminated.Message
+								} else if !gotPodLogs {
+									gotPodLogs = true
```

Comment:
Are tasks parallelized? Does that mean we should be using something like a `sync.Once` rather than a bool?

Reply:
`statusTask` are [run concurrently using a workerpool](https://github.com/cilium/cilium-cli/blob/0d9bb5cacf64b162c70a78e5bd4b40d33fac5b4d/status/k8s.go#L565-L569) so I agree that synchronization is required here.

Reply:
You caught me :-p. I figured the worst case was that we downloaded logs twice. But it could also make the race detector eventually scream, so I'll fix it.